tezos-sig-gating-contracts/scripts/tests/gatednftminter_nodispatch.spec.ts

```diff
@@ -0,0 +1,719 @@
+  let currentChainId: string;
+  let nexeraSignerPublicKey: string;
+
+  before(async () => {
+    // SET SIGNER
+    deployerAddress = "tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb";
+    Tezos.setProvider({
+      signer: await InMemorySigner.fromSecretKey(
+        "edsk3QoqBuvdamxouPhin7swCvkQNgq4jP5KZPbwWNnwdZpSpJiEbq"
+      ),
+    });
+    // Retrieve Signer public key
+    nexeraSignerPublicKey = await nexeraSigner.publicKey();
+    // Retrieve the chainID
+    currentChainId = await client.getChainId();
+    // DEPLOY NFTMINTER
+    exampleGatedNFTMinter = await deployNFTMinterExtNoDispatch(Tezos);
+  });
+
+  beforeEach(async () => {
+    const block = await client.getBlockHeader();
+    currentBlock = block.level;
+  });
+
+  it(`Check initial storage (the deployer is the admin of NftMinter and owns the asset #0)`, async () => {
+    const cntr = await Tezos.contract.at(
+      exampleGatedNFTMinter ? exampleGatedNFTMinter : ""
+    );
+    const storage: any = await cntr.storage();
+    // Verify
+    const admin = await storage.admin;
+    const asset0_owner = await storage.siggated_extension.ledger.get(0);
+    const asset1_owner = await storage.siggated_extension.ledger.get(1);
+    const asset2_owner = await storage.siggated_extension.ledger.get(2);
+    const asset3_owner = await storage.siggated_extension.ledger.get(3);
+    expect(deployerAddress === admin).to.be.true;
+    expect(asset0_owner === deployerAddress).to.be.true;
+    expect(asset1_owner).to.be.undefined;
+    expect(asset2_owner).to.be.undefined;
+    expect(asset3_owner).to.be.undefined;
+  });
+
+  it(`Should mint the asset #1`, async () => {
+    // Get contract storage
+    const cntr = await Tezos.contract.at(
+      exampleGatedNFTMinter ? exampleGatedNFTMinter : ""
+    );
+
+    // MINT OFFCHAIN
+    const functionCallContract = exampleGatedNFTMinter
+      ? exampleGatedNFTMinter
+      : "";
+    const functionCallArgs = {
+      owner: "tz1fon1Hp3eRff17X82Y3Hc2xyokz33MavFF",
+      token_id: "1",
+    };
+    // Prepare Hash of payload
+    const functionCallArgsBytes = convert_mint(
+      functionCallArgs.owner,
+      functionCallArgs.token_id
+    );
+    const payloadToSign: TezosTxAuthData = {
+      chainID: currentChainId,
+      userAddress: "tz1fon1Hp3eRff17X82Y3Hc2xyokz33MavFF",
+      nonce: 0,
+      blockExpiration: currentBlock + 10,
+      contractAddress: functionCallContract,
+      functionCallName: "%mint_gated",
+      functionCallArgs: functionCallArgsBytes,
+      signerPublicKey: nexeraSignerPublicKey,
+    };
+    const payloadHash = computePayloadHash(payloadToSign);
+    // Nexera signs Hash of payload
+    let signature = await nexeraSigner.sign(payloadHash);
+    // Execute mint-offchain entrypoint
+    const args: TezosTxCalldata = buildTxInputFromTxAuthData(
+      payloadToSign,
+      signature.prefixSig
+    );
+    // CALL contract
+    const op = await cntr.methodsObject.exec_gated_calldata(args).send();
```

Comment:
So ideally I'd like to see an example that wraps exec_gated_calldata(args) like this:

```
mintNFT(args,signature){
 dispatch({
  "mintNFT",
  args,
  signature
  ```

Reply:
I'll add other examples in another PR